src/util/index_tree.ts

```diff
@@ -401,6 +434,15 @@ export abstract class IndexTreeNode<T extends IndexTreeNode<T>> {
       throw new Error('Text node cannot have children');
     }
 
+    if (node.isRemoved) {
+      const index = this._children.indexOf(node);
+      const refined =
+        [...this._children].splice(0, index).filter((node) => !node.isRemoved)
+          .length - 1;
+
+      return refined < 0 ? 0 : refined;
+    }
```

Comment:
Is this part dealing with the deletion of concurrent editing situations?

Reply:
it's for filtering tombstone nodes when calcualting index/path for changes